### DISCUSSION TITLE
How to calculate daily returns

### DISCUSSION THREAD

**Discussion by linmingren** [nautechsystems/nautilus_trader#776] [Q&A] [answered]
I had run a backtest with 6 days bar data, but I noticed that there is only one  'returns' value in method ReturnsVolatility#calculate_from_returns. Shouldn't this parameter have 6 elements?

**Comment by cjdsellers** [comment, marked as answer]
Hi @linmingren 

Thanks for the question. So the built in returns statistics calculate per position open to close, rather than any particular time period. This is something which we could add later, you could also calculate this with some logic in your strategy

**Reply by linmingren** [reply]
I don't have a good idea about how to determine if day changed, the only idea comes in my mind is something like below:

on_bar(Bar):
    check if day changed according to the bar time, calculate the return value of previous day if day changed and store the value to cache.

**Reply by cjdsellers** [reply]
yes exactly, something like that or some variation
